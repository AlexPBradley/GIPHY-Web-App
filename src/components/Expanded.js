import React from "react";
import "../styles/expanded.css";

const Expanded = ({ selected, setIsSelected }) => {
  return (
    <div className="expanded">
      <div className="background" onClick={() => setIsSelected()}></div>
        <img
          className="gif"
          data-testid="gif"
          src={selected?.images.downsized.url}
          alt="Gif"
        />
        <div className="test">{selected.title}</div>        
    </div>
  );
};

export default Expanded;
