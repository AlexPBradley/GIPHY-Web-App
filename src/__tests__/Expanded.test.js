import React from "react";
import { render, screen } from "@testing-library/react";
import Expanded from "../components/Expanded";

describe("Expanded", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Expanded />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("expanded gif renders", () => {
    render(<Expanded />);

    const gif = screen.getByTestId("gif");

    expect(gif).toBeInstanceOf(HTMLImageElement);
  });

  it("title renders", () => {
    render(<Expanded />);

    const expandedTitle = screen.getByTestId("expanded-title");

    expect(expandedTitle).toBeInstanceOf(HTMLDivElement);
  });
});
