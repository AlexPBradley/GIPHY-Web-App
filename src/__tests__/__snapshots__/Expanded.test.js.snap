// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expanded renders correctly 1`] = `
<DocumentFragment>
  <div
    class="expanded"
  >
    <div
      class="background"
    />
    <img
      alt="Gif"
      class="gif"
      data-testid="gif"
    />
    <div
      class="expanded-title"
      data-testid="expanded-title"
    />
  </div>
</DocumentFragment>
`;
